proxy/reverse.go

```diff
@@ -118,6 +121,9 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 		redirectRequest(proxyreq, ep.URL)
 
 		res, err = p.transport.RoundTrip(proxyreq)
+		if requestCanceled {
```

Comment:
When we cancel the request after it is already finished and we are past the requestCanceled check than the err value  will only indicate if the proxy request itself was ok or not and we will fail when we write the response to the client.